src/test/ui/lint/must_not_suspend/dedup.stderr

```diff
@@ -0,0 +1,19 @@
+error: `No` held across a suspend point, but should not be
+  --> $DIR/dedup.rs:16:12
+   |
+LL |     wheeee(No {}).await;
+   |     -------^^^^^------- the value is held across this suspend point
+   |
+note: the lint level is defined here
+  --> $DIR/dedup.rs:3:9
+   |
+LL | #![deny(must_not_suspend)]
+   |         ^^^^^^^^^^^^^^^^
+help: consider using a block (`{ ... }`) to shrink the value's scope, ending before the suspend point
+  --> $DIR/dedup.rs:16:12
+   |
+LL |     wheeee(No {}).await;
+   |            ^^^^^
```

Comment:
this only makes sense where the value is held locally. If it was found in anther generator, then we should probably not emit this help

Reply:
this is triggering on the `No { }` value being held across the `whee` yield point, I think? I believe that @eholk is improving this analysis